src/hooks/api/getters/useEvents/useEvents.ts

```diff
@@ -0,0 +1,52 @@
+import useSWR, { mutate, SWRConfiguration } from 'swr';
+import { useCallback } from 'react';
+import { formatApiPath } from '../../../../utils/format-path';
+import handleErrorResponses from '../httpErrorResponseHandler';
+import { IEvent } from '../../../../interfaces/event';
+
+const PATH = formatApiPath('api/admin/events');
+
+export interface UseEventsOutput {
+    events: IEvent[];
+    refetchEvents: () => void;
+    loading: boolean;
+    error?: Error;
+}
+
+export const useEvents = (
```

Comment:
I'm wondering if this could be easier represented as two separate hooks: 

```
useEvents
useFeatureEvents
```

I think what you have done here is cool, but I had to spend some time going through what was happening. It would be easier to reason about if they were separate hooks, and perhaps also easier to understand what was happening in the context they were being used. What do you think?

Reply:
Yeah, that's much better! Will fix.